### ISSUE TITLE
REST-assured requests are not logged by Serenity if they have .log()... in its chain

### ISSUE THREAD

**Issue by dsmt** [serenity-bdd/serenity-core#185]
This request won't be logged by Serenity due to `.log().all()`

    rest()
                .contentType("application/json; charset=utf-8")
                .header(getAuthHeader())
                .body(shipmentJson)
                .log().all()
                .post(System.getProperty("backend.url") + "/api/shipments");

**Comment by YamStranger** [comment]
Hello @dsmt. Thanks for this issue!

In serenity we use internally [rest-assured](https://github.com/jayway/rest-assured). We add request/response information only in aggregation reports, logging of body, headers, cookies, and so on already implemented in it [rest-assured](https://github.com/jayway/rest-assured) and can be enabled with some configuration according to their [manual](https://github.com/jayway/rest-assured/wiki/Usage).  

Here you can find all necessary information about [rest-assured Logging](https://github.com/jayway/rest-assured/wiki/Usage#logging) 

P.S. By default .log().... prints to stdout, so if you run your build (tets) with --info you find all necessary info.

**Comment by dsmt** [comment]
Well, the problem with log().... is when I add it to the step all rest-assured log disappearing from Serenity report.

I do have rest-assured log in stdout:

![](http://take.ms/Ty9Vf)

But Serenity report missing inner REST-step

![](http://take.ms/uL6cu)

**Comment by YamStranger** [comment]
@dsmt thanks for excellent explanation, it speed up investigation very much!!!!

It will be fixed as soon as it possible, for now if it possible please try to use workaround:

code to reproduce this issue:
```
         rest().given().contentType("application/json")
                .content(jsonPet).log().all().post("http://petstore.swagger.io/v2/pet");
```

workaround:
```
        RequestSpecification spec = rest().given().contentType("application/json")
                .content(jsonPet);
        spec.log().all();
        spec.post("http://petstore.swagger.io/v2/pet");
```
 
Thanks!!!

**Comment by YamStranger** [comment]
Hello @dsmt !

This issue should be fixed under [1.1.22-rc.11](https://github.com/serenity-bdd/serenity-core/releases/tag/v1.1.22-rc.11), please check it. 

If you feel that this issue already fixed and cant be reproduced - please close this issue. 

Thanks a lot!!

**Comment by dsmt** [comment]
Thanks for fixing, it works pretty well for me =)
